hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/BaseActionExecutor.java

```diff
@@ -132,4 +145,22 @@ protected final void writeTableMetadata(HoodieRestoreMetadata metadata) {
       }
     }
   }
+
+  /**
+   * Drop indexes information, for e.g., restore operation.
+   */
+  protected final void dropIndexInfo() {
+    StoragePath metadataTableBasePath =
+        HoodieTableMetadata.getMetadataTableBasePath(table.getMetaClient().getBasePath());
+    List<String> partitionPaths = FSUtils.getAllPartitionPaths(context, table.getStorage(), metadataTableBasePath, false);
+    for (String partitionPath : partitionPaths) {
+      if (FUNCTIONAL_INDEX != MetadataPartitionType.fromPartitionPath(partitionPath)
+          && SECONDARY_INDEX != MetadataPartitionType.fromPartitionPath(partitionPath)) {
+        HoodieTableMetadataUtil.deleteMetadataTablePartition(
+            table.getMetaClient(), context, partitionPath, true);
+      } else {
+        // Delete records in functional and secondary indexes, but keep the definitions.
```

Comment:
we can get the index definition from metaClient.getIndexMetadata()

Reply:
not needed here.. added a method in `MetadataPartitionType` which tells whether index needs to be deleted for restore or not.